### PR TITLE
log analytics

### PR DESCRIPTION
**Pull request by stishkin**

### CODE REVIEW

File: src/ApiService/ApiService/OneFuzzTypes/Events.cs [hunk 1]
```diff
@@ -180,11 +180,11 @@ Guid PingId
 //    ) : BaseEvent();
 
 
-//record EventScalesetResizeScheduled(
-//    Guid ScalesetId,
-//    PoolName PoolName,
-//    int size
-//    ) : BaseEvent();
+public record EventScalesetResizeScheduled(
```
Comment (tevoinea) [inline src/ApiService/ApiService/OneFuzzTypes/Events.cs:183]: Don't forget to add these to `GetEventType` and `GetTypeInfo`

File: src/ApiService/ApiService/OneFuzzTypes/Model.cs [hunk 2]
```diff
@@ -513,8 +514,26 @@ public record Pool(
 ) : StatefulEntityBase<PoolState>(State);
 
 
-// TODO
-public record AgentConfig();
+public record ClientCredentials
+(
+    Guid ClientId,
+    string ClientSecret
```
Comment (nharper285) [inline src/ApiService/ApiService/OneFuzzTypes/Model.cs:520]: How are we protecting this value? Or, maybe a better question, are we making sure this doesn't get logged anywhere?

File: src/ApiService/ApiService/onefuzzlib/Containers.cs [hunk 3]
```diff
@@ -108,17 +129,48 @@ public class Containers : IContainers {
...
     }
 
-    public async System.Threading.Tasks.Task saveBlob(Container container, string name, string data, StorageType storageType) {
+    public async Async.Task SaveBlob(Container container, string name, string data, StorageType storageType) {
         var client = await FindContainer(container, storageType) ?? throw new Exception($"unable to find container: {container.ContainerName} - {storageType}");
-
         await client.UploadBlobAsync(name, new BinaryData(data));
     }
 
+    //TODO: get this ones on startup and cache (and make this method un-accessible to everyone else)
```
Comment (nharper285) [inline src/ApiService/ApiService/onefuzzlib/Containers.cs:137]: Did you want to make GetInstanceId private?

File: src/ApiService/ApiService/onefuzzlib/LogAnalytics.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        var client = new OperationalInsightsManagementClient(new Rest.TokenCredentials(token.Token)) { SubscriptionId = _creds.GetSubcription() };
+        var customerId = (await client.Workspaces.ListByResourceGroupAsync(_creds.GetBaseResourceGroup()))
+                        .Select(w => w.CustomerId)
+                        .First();
+        var keys = await client.SharedKeys.GetSharedKeysAsync(_creds.GetBaseResourceGroup(), _config.OneFuzzMonitor);
+        return new MonitorSettings(customerId, keys.PrimarySharedKey);
+    }
+
+
+    public ResourceIdentifier GetWorkspaceId() {
```
Comment (nharper285) [inline src/ApiService/ApiService/onefuzzlib/LogAnalytics.cs:36]: Looks like `GetSubcription()` is misspelled. Can we fix that with this PR.

File: src/ApiService/ApiService/onefuzzlib/Containers.cs (hunk 3 above)
Comment (stishkin) [reply src/ApiService/ApiService/onefuzzlib/Containers.cs:137]: that's for later

File: src/ApiService/ApiService/OneFuzzTypes/Model.cs (hunk 2 above)
Comment (stishkin) [reply src/ApiService/ApiService/OneFuzzTypes/Model.cs:520]: same behaviour as in python code

**Review by tevoinea — APPROVED** [review]

**Pull request merged**